### PR TITLE
Document how to build fully static native executables

### PR DESCRIPTION
**Pull request by andreaTP**
Following up on this discussion:
https://groups.google.com/g/quarkus-dev/c/aYJQjXyVlpc/m/AYQTEPheAgAJ

Here you have minimal documentation for building fully static binaries and Docker images from `scratch`.

### CODE REVIEW

**Review by cescoffier — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/building-native-image.adoc [hunk 1]
```diff
@@ -227,6 +227,11 @@ To produce a native executable, this means that the `--enable-preview` flag need
 You can do so by prepending the flag with `-J` and passing it as additional native build argument: `-Dquarkus.native.additional-build-args=-J--enable-preview`.
 ====
 
+=== Build fully static native executables
+
+On Linux it's possible to package a native executable that doesn't depend on any system shared library.
+There are https://www.graalvm.org/22.1/reference-manual/native-image/StaticImages/#preparation[some system requirements] to be fulfilled and additional build arguments to be used along with the `native-image` invocation, a minimum is `-Dquarkus.native.additional-build-args="--static","--libc=musl"`.
+
```
[the remark this answers is not in the document]
Comment (andreaTP) [reply docs/src/main/asciidoc/building-native-image.adoc:236]: Done here: https://github.com/quarkusio/quarkus/pull/25890/commits/0057fd328a0d077682edfc440e6dc908a4dfa382
let me know if you prefer a different wording.

**Comment by andreaTP** [comment]
@cescoffier thanks a lot for your feedback!

For context, I have been shipping production applications as fully statically linked binaries for quite some time now, in practice, I have never faced issues caused by the swap `musl` / `glibc`.

Examples are in:
- [`codacy-coverage-reporter`](https://github.com/codacy/codacy-coverage-reporter) switched to fully static binaries to enable running the binary on "stripped down" Alpine Docker images [reference](https://github.com/codacy/codacy-coverage-reporter/issues/212)
- [`kubectl-clowdflow`](https://github.com/lightbend/cloudflow) has been shipped as a fully static binary since day 0 and I [blogged about](https://www.lightbend.com/blog/writing-kubectl-plugins-with-scala-or-java-with-fabric8-kubernetes-client-on-graalvm)

> It would be interesting to try with that approach (with the words of caution I mentioned in my review).

Do I understand correctly that you would like to experiment with statically linking `glibc`? In this case, it looks like there are serious issues in doing that: https://stackoverflow.com/a/57478728
I noticed that they added support for Bionic LibC if there is any interest in it.

Last note, this documentation improvement comes after my experiments for statically linking using `zig cc` as a drop-in replacement for `gcc` which dramatically simplifies the setup for static linking [reference](https://github.com/andreaTP/quarkus_fully_static/blob/main/Dockerfile.zig).

**Comment by cescoffier** [comment]
We have seen issues especially when your application uses native libraries (which is not as rare as people thinks).

About distroless, I was thinking about using your approach and removing glibc from the base image (as an alternative). I believe it should work for most applications (except with the limits I mentioned above)

**Comment by andreaTP** [comment]
Got it, thanks for explaining!
Regarding distroless it should work out of the box 👍

**Comment by andreaTP** [comment]
@cescoffier I'm not an expert in distroless images but changing `FROM scratch` to `FROM gcr.io/distroless/static` generates a working container image (tested only with a basic hello world program).

File: docs/src/main/asciidoc/building-native-image.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (cescoffier) [reply docs/src/main/asciidoc/building-native-image.adoc:236]: thanks!

**Review by cescoffier — APPROVED** [review]

**Comment by cescoffier** [comment]
Can you squash your commits?

**Comment by andreaTP** [comment]
Sure!

**Comment by andreaTP** [comment]
@cescoffier done, thanks for the review!

**Comment by cescoffier** [comment]
Awesome! Thanks!

**Pull request merged**